Component argument of `Option<Callback<(), bool>>` is unsupported in main, but `Option<EventHandler<()>>` works

## Feature Request

With great thanks to @ealmloff for https://github.com/DioxusLabs/dioxus/pull/2437 (!!), it seems that there is one feature that did not come automatically with those changes: support for `Option<Callback<T, R>>` in component args.

This compiles:
```rust
#[component]
fn Comp(cb: Option<EventHandler<()>>) -> Element {
    rsx! {
        div {
            onclick: move |_| {
                if cb.is_some() { cb.unwrap()(()) };
            },
            "Click"
        }
    }
}

#[component]
fn Home() -> Element {
    rsx! {
        Comp {
            cb: move |_| {
                tracing::info!("Clicked");
            },
        }
    }
}
```

This fails with a compiler error:
```rust
#[component]
fn Comp(cb: Option<Callback<(), bool>>) -> Element {
    rsx! {
        div {
            onclick: move |_| {
                if cb.is_some() { cb.unwrap()(()) } else { false };
            },
            "Click"
        }
    }
}

#[component]
fn Home() -> Element {
    rsx! {
        Comp {
            cb: move |_| {
                tracing::info!("Clicked");
                true
            },
        }
    }
}
```

Error:
```
 error[E0277]: the trait bound `Option<dioxus::prelude::Callback<(), bool>>: From<{closure@src/main.rs:62:17: 62:25}>` is not satisfied
  --> src/main.rs:62:17
   |
62 |               cb: move |_| {
   |  _____________--__^
   | |             |
   | |             required by a bound introduced by this call
63 | |                 tracing::info!("Clicked");
64 | |                 true
65 | |             },
   | |_____________^ the trait `From<{closure@src/main.rs:62:17: 62:25}>` is not implemented for `Option<dioxus::prelude::Callback<(), bool>>`, which is required by `{closure@src/main.rs:62:17: 62:25}: SuperInto<Option<dioxus::prelude::Callback<(), bool>>, _>`
```

## Implement Suggestion

I looked but could not find where the `From<Option<EventHandler>>` would be defined so as to simply put up a PR. If I could be given direction where to look, I'm happy to attempt this myself.

--- a/packages/core/src/properties.rs
+++ b/packages/core/src/properties.rs
@@ -243,13 +243,19 @@ impl<'a> SuperFrom<Arguments<'a>, OptionArgumentsFromMarker> for Option<String> 
 }
 
 #[doc(hidden)]
-pub struct OptionHandlerMarker;
+pub struct OptionCallbackMarker<T>(std::marker::PhantomData<T>);
 
-impl<G: 'static, F: FnMut(G) + 'static> SuperFrom<F, OptionHandlerMarker>
-    for Option<EventHandler<G>>
+// Closure can be created from FnMut -> async { anything } or FnMut -> Ret
+impl<
+        Function: FnMut(Args) -> Spawn + 'static,
+        Args: 'static,
+        Spawn: SpawnIfAsync<Marker, Ret> + 'static,
+        Ret: 'static,
+        Marker,
+    > SuperFrom<Function, OptionCallbackMarker<Marker>> for Option<Callback<Args, Ret>>
 {
-    fn super_from(input: F) -> Self {
-        Some(EventHandler::new(input))
+    fn super_from(input: Function) -> Self {
+        Some(Callback::new(input))
     }
 }
 

--- a/packages/core/src/properties.rs
+++ b/packages/core/src/properties.rs
@@ -261,6 +261,21 @@ impl<
 
 #[test]
 #[allow(unused)]
+fn optional_callback_compiles() {
+    fn compiles() {
+        // Converting from closures (without type hints in the closure works)
+        let callback: Callback<i32, i32> = (|num| num * num).super_into();
+        let callback: Callback<i32, ()> = (|num| async move { println!("{num}") }).super_into();
+
+        // Converting from closures to optional callbacks works
+        let optional: Option<Callback<i32, i32>> = (|num| num * num).super_into();
+        let optional: Option<Callback<i32, ()>> =
+            (|num| async move { println!("{num}") }).super_into();
+    }
+}
+
+#[test]
+#[allow(unused)]
 fn from_props_compiles() {
     // T -> T works
     let option: i32 = 0i32.super_into();

